### PR TITLE
splitting, ts & cleanup (preparing for graph optimization)

### PR DESCRIPTION
**Pull request by JulianWielga**
I believe that there is no change in logic here.

### CODE REVIEW

File: ui/client/components/graph/NodeUtils.ts [hunk 1]
```diff
@@ -1,32 +1,62 @@
...
+type Process = {
+  nodes: Node[],
+  edges: Edge[],
+  properties: $TodoType,
+}
+
 class NodeUtils {
 
-  isNode = (obj) => {
+  isNode = (obj: {}): obj is Node => {
```
Comment (lciolecki) [inline ui/client/components/graph/NodeUtils.ts]: I know it's refactor.. But can we do that kind of functions do as arrow function in one line? It looks much better.. :)

File: ui/client/components/graph/NodeUtils.ts [hunk 2]
```diff
@@ -165,21 +201,27 @@ class NodeUtils {
...
       (groups) => (groups || []).map(group => ({
         ...group,
         nodes: nodeOperation(group.nodes),
       })),
-      processToDisplay)
+      processToDisplay,
+    )
   }
 
   _canHaveMoreInputs = (nodeTo, nodeInputs, processDefinitionData) => {
```
Comment (lciolecki) [inline ui/client/components/graph/NodeUtils.ts]: No we cane remove that prefix `_` for private / protected method..

File: ui/client/components/graph/NodeUtils.ts [hunk 3]
```diff
@@ -1,32 +1,62 @@
+/* eslint-disable i18next/no-literal-string */
 import _ from "lodash"
```
Comment (lciolecki) [inline ui/client/components/graph/NodeUtils.ts]: As you have saide - we should import only used method from lodash ;)

File: ui/client/components/graph/Graph.js [hunk 4]
```diff
@@ -152,39 +167,23 @@ class Graph extends React.Component {
     return this.state.exported
   }
 
-  validateConnection = (cellViewS, magnetS, cellViewT, magnetT) => {
-    const from = cellViewS.model.id
-    const to = cellViewT.model.id
-    return magnetT && NodeUtils.canMakeLink(from, to, this.props.processToDisplay, this.props.processDefinitionData)
+  validateConnection() {
```
Comment (mproch) [inline ui/client/components/graph/Graph.js]: why is it better than previous version? I'm just curious ;)

Comment (JulianWielga) [reply ui/client/components/graph/Graph.js]: I try to separate component props from inner joint/paper args

File: ui/client/components/graph/NodeUtils.ts [hunk 5]
```diff
@@ -194,32 +236,41 @@ class NodeUtils {
...
     return this.edgesFromProcess(process).filter(e => e.from == nodeId)
   }
 
-  edgeId = (edge) => {
+  edgeId = (edge: Edge): string => {
     return `${edge.from}-${edge.to}`
   }
 
-  groupIncludesOneOfNodes = (nodeGroup, nodeIds)  => _.some(nodeGroup.nodes, (node) => _.includes(nodeIds, _.get(node, "id") || node))
+  groupIncludesOneOfNodes = (nodeGroup, nodeIds: NodeId[]) => _.some(
```
Comment (JulianWielga) [inline ui/client/components/graph/NodeUtils.ts]: something is wrong here. I can't identify the nodeGroup type.

File: ui/client/test/ValidationLabels-test.js [hunk 6]
```diff
@@ -43,9 +42,7 @@ describe("test validation labels", () => {
 
       //when
       render(
-        <I18nextProvider i18n={i18n}>
-          <ValidationLabels validators={validators} values={[""]}/>
-        </I18nextProvider>
+        <ValidationLabels validators={validators} values={[""]}/>
```
Comment (lciolecki) [inline ui/client/test/ValidationLabels-test.js]: Are you sure, doest it work after remove `I18nextProvider`?

**Pull request closed without merging**